charts/db-backup/scripts/backup-postgres

```diff
@@ -31,10 +31,17 @@ restore () {
   if [[ "$GOVUK_ENVIRONMENT" = *"prod"* ]]; then
     : "${REALLY_RESTORE_ONTO_PRODUCTION?}"
   fi
-  local s3_url="$BUCKET/$PGHOST/$FILENAME"
+  local s3_url="$BUCKET/$DB_HOST/$FILENAME"
   s3_url="$s3_url" dump_is_readable
-  dropdb -ef --if-exists "$PGDATABASE"
-  aws s3 cp "$s3_url" - | progress | pg_restore -Cd postgres --no-comments
+
+  dropdb -ef --if-exists "$DB_DATABASE-restore"
+  createdb -eT template0 "$DB_DATABASE-restore"
+  aws s3 cp "$s3_url" - | progress | pg_restore -d "$DB_DATABASE-restore" --no-comments
+  psql -c "ALTER DATABASE \"$DB_DATABASE-restore\" OWNER TO \"$DB_OWNER\";" || true
+
+  psql -c "ALTER DATABASE \"$DB_DATABASE\" RENAME TO \"$DB_DATABASE-old\";" || true
+  psql -c "ALTER DATABASE \"$DB_DATABASE-restore\" RENAME TO \"$DB_DATABASE\";"
```

Comment:
Would it be safer to combine these two into one `-c "..."` block, and therefore implicitly into one transaction? That way even if something goes wrong, you're still going to be left with a usable original database. The risk is small, admittedly, but maybe worth pre-emptively handling. 

```suggestion
  psql -c "ALTER DATABASE \"$DB_DATABASE\" RENAME TO \"$DB_DATABASE-old\"; ALTER DATABASE \"$DB_DATABASE-restore\" RENAME TO \"$DB_DATABASE\";"
```

Reply:
I may eat my words if one of the really slow ones gets nobbled by some client query right at the end 😂

Reply:
Totally makes sense, happy to go with this as-is - thanks! :rocket: